Parse errors occur on formulas with `remark-math`

### Initial checklist

- [x] I read the [support docs](https://mdxjs.com/community/support/)
- [x] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [x] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/main/code-of-conduct.md)
- [x] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=issues) and [discussions](https://github.com/orgs/mdx-js/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

eslint-plugin-mdx@3.4.1

### Steps to reproduce

A minimal and reproducible repository is available here:

[https://github.com/r4ai/eslint-plugin-mdx-math-issue](https://github.com/r4ai/eslint-plugin-mdx-math-issue)

### Actual behavior

I added `remark-math` to the `plugins` array in `.remarkrc.js` as shown below, but the parse error still persists.

`.remarkrc.js`:

```js
import remarkPresetLintConsistent from "remark-preset-lint-consistent";
import remarkPresetLintRecommended from "remark-preset-lint-recommended";
import remarkMath from "remark-math";
import remarkPresetLintMarkdownStyleGuide from "remark-preset-lint-markdown-style-guide";
import remarkPresetPrettier from "remark-preset-prettier";

export default {
  plugins: [
    remarkMath,
    remarkPresetLintConsistent,
    remarkPresetLintRecommended,
    remarkPresetLintMarkdownStyleGuide,
    remarkPresetPrettier,
  ],
};
```

`math.mdx`:

```sh
$$
\forall \epsilon > 0, \exists \delta > 0 \text{ s.t. } |x - a| < \delta \Rightarrow |f(x) - b| < \epsilon
$$
```

```sh
$ bun run eslint .

/home/r4ai/src/repos/github.com/r4ai/eslint-plugin-mdx-math/src/math.mdx
  0:54  error  Preprocessing error: Could not parse expression with acorn

✖ 1 problem (1 error, 0 warnings)
```

### Expected behavior

Parse normally

### Runtime

node@22.15.0

### Package manager

npm@10.9.2

### Operating system

Ubuntu 24.04.2 LTS

### Build and bundle tools

_No response_

--- a/.changeset/sour-icons-complain.md
+++ b/.changeset/sour-icons-complain.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-mdx": patch
+---
+
+fix: use unified `performSyncWork` for parsing with remark config

--- a/package.json
+++ b/package.json
@@ -51,6 +51,7 @@
     "path-serializer": "^0.5.0",
     "prettier": "^3.5.3",
     "react": "^19.1.0",
+    "remark-math": "^6.0.0",
     "rimraf": "^6.0.1",
     "simple-git-hooks": "^2.13.0",
     "ts-node": "^10.9.2",
@@ -102,7 +103,8 @@
   "prettier": "@1stg/prettier-config",
   "remarkConfig": {
     "plugins": [
-      "@1stg/preset"
+      "@1stg/preset",
+      "remark-math"
     ]
   },
   "typeCoverage": {

--- a/packages/eslint-plugin-mdx/src/from-markdown.ts
+++ b/packages/eslint-plugin-mdx/src/from-markdown.ts
@@ -1,9 +1,0 @@
-import type { Root } from 'mdast'
-import type { Value } from 'mdast-util-from-markdown'
-import { createSyncFn } from 'synckit'
-
-import { cjsRequire } from './helpers.js'
-
-export const fromMarkdown = createSyncFn<
-  (value: Value, isMdx: boolean) => Root
->(cjsRequire.resolve('./worker.js'))

--- a/packages/eslint-plugin-mdx/src/processors/markdown.ts
+++ b/packages/eslint-plugin-mdx/src/processors/markdown.ts
@@ -1,10 +1,13 @@
 /** {@link https://github.com/eslint/markdown/blob/f17311eb250fe3f41a24fd52be9c6713d12b7d67/src/processor.js} */
 
+import fs from 'node:fs'
+import path from 'node:path'
+
 import type { AST, Linter, Rule } from 'eslint'
+import { performSyncWork } from 'eslint-mdx'
 import type { Node, Parent, Nodes, Root, RootContentMap } from 'mdast'
 import type { MdxFlowExpression, MdxTextExpression } from 'mdast-util-mdx'
 
-import { fromMarkdown } from '../from-markdown.js'
 import { meta } from '../meta.js'
 
 import type { CodeBlock, RangeMap } from './types.js'
@@ -263,21 +266,42 @@ function fileNameFromMeta(block: CodeBlock) {
 }
 
 /**
+ * Given a filepath, get the nearest path that is a regular file. The filepath
+ * provided by eslint may be a virtual filepath rather than a file on disk. This
+ * attempts to transform a virtual path into an on-disk path
+ */
+function getOnDiskFilepath(filepath: string): string {
+  let fallback: boolean | undefined
+  try {
+    /* istanbul ignore next -- `throwIfNoEntry` compatibility */
+    if (!fs.statSync(filepath, { throwIfNoEntry: false })) {
+      fallback = true
+    }
+  } catch (err) {
+    /* istanbul ignore next -- https://github.com/eslint/eslint/issues/11989 */
+    if ((err as NodeJS.ErrnoException).code === 'ENOTDIR') {
+      fallback = true
+    }
+  }
+  return fallback ? getOnDiskFilepath(path.dirname(filepath)) : filepath
+}
+
+/**
  * Extracts lintable code blocks from Markdown text.
  *
  * @param sourceText The text of the file.
  * @param filename The filename of the file
  * @returns Source code blocks to lint.
  */
-
 function preprocess(sourceText: string, filename: string) {
   // istanbul ignore next
   const text = sourceText.startsWith(BOM) ? sourceText.slice(1) : sourceText
-  const ast = fromMarkdown(
-    text,
+  const { root } = performSyncWork({
+    filePath: getOnDiskFilepath(filename),
+    code: text,
     // FIXME: how to read `extensions` and `markdownExtensions` parser options?
-    filename.endsWith('.mdx'),
-  )
+    isMdx: filename.endsWith('.mdx'),
+  })
   const blocks: CodeBlock[] = []
 
   blocksCache.set(filename, blocks)
@@ -300,7 +324,7 @@ function preprocess(sourceText: string, filename: string) {
     }
   }
 
-  traverse(ast, {
+  traverse(root, {
     '*'() {
       allComments = []
     },

--- a/test/fixtures/576.mdx
+++ b/test/fixtures/576.mdx
@@ -1,0 +1,5 @@
+$$
+\forall \epsilon > 0, \exists \delta > 0 \text{ s.t. } |x - a| < \delta \Rightarrow |f(x) - b| < \epsilon
+$$
+
+the aggregate amount would be $100Mm*10^{18}$

